### ISSUE TITLE
Test with the reprocessed image

### ISSUE THREAD

**Issue by WeicongChen** [sarathknv/adversarial-examples-pytorch#1]
Hi, great repo!

I am little confused with this line 
https://github.com/sarathknv/adversarial-examples-pytorch/blob/54887aa583ee19e94fc95e23684271019eecdd68/fgsm.py#L84
As far as I know, it is more reasonable to test with the reprocessed image rather than `inp`.

**Comment by sarathknv** [comment]
By 'reprocessed image'  do you mean image with preprocessing undone?

**Comment by juicecwc** [comment]
No, what I mean is the adversarial image. Better viewed in the following code.
```python
adv = inp.data.cpu().numpy()[0]
adv = adv.transpose(1, 2, 0)
adv = (adv * std) + mean
adv = np.clip(adv, 0, 1)
adv = (adv - mean) / std # this is what i mean
```

**Comment by sarathknv** [comment]
You are clipping the adversarial image before inputting to the model, so that after undoing preprocessing it is within the `[0 255]` bound.
Yeah, this is sensible.
